spec/integration/aws_network_acl_spec.rb

```diff
@@ -0,0 +1,107 @@
+              { :rule_number=>100, :protocol=>"-1", :rule_action=>"deny", :egress=>false, :cidr_block=>"10.0.0.0/24" },
+              { :rule_number=>200, :protocol=>"-1", :rule_action=>"allow", :egress=>false, :cidr_block=>"0.0.0.0/0" },
+              { :rule_number=>300, :protocol=>"6", :rule_action=>"allow", :egress=>false, :cidr_block=>"172.31.0.0/22", :port_range=>{ :from=>22, :to=>23 } },
+              { :rule_number=>32767, :protocol=>"-1", :rule_action=>"deny", :egress=>false, :cidr_block=>"0.0.0.0/0" }
+            ]
+        ).and be_idempotent
+      end
+
+      context 'when rules are empty' do
+        aws_network_acl 'test_network_acl' do
+          vpc 'test_vpc'
+          inbound_rules(rule_number: 100, action: :deny, protocol: -1, cidr_block: '10.0.0.0/24')
+          outbound_rules(rule_number: 500, action: :allow, protocol: -1, cidr_block: '0.0.0.0/0')
+        end
+
+        it "aws_network_acl 'test_network_acl' removes current rules" do
+          expect_recipe {
+            aws_network_acl 'test_network_acl' do
+              vpc 'test_vpc'
+              inbound_rules []
+              outbound_rules []
+            end
+          }.to create_an_aws_network_acl('test_network_acl',
+            vpc_id: test_vpc.aws_object.id,
+            entry_set:
+              [
+                { :rule_number=>32767, :protocol=>"-1", :rule_action=>"deny", :egress=>true, :cidr_block=>"0.0.0.0/0" },
+                { :rule_number=>32767, :protocol=>"-1", :rule_action=>"deny", :egress=>false, :cidr_block=>"0.0.0.0/0" }
+              ]
+          ).and be_idempotent
+        end
+      end
+
+      context 'when rules are nil' do
+        aws_network_acl 'test_network_acl' do
+          vpc 'test_vpc'
+          inbound_rules(rule_number: 100, action: :deny, protocol: -1, cidr_block: '10.0.0.0/24')
+          outbound_rules(rule_number: 500, action: :allow, protocol: -1, cidr_block: '0.0.0.0/0')
+        end
+
+        it "aws_network_acl 'test_network_acl' with a nil rules array leaves current rules alone" do
+          expect_recipe {
+            aws_network_acl 'test_network_acl' do
+              vpc 'test_vpc'
+              inbound_rules nil
+              outbound_rules nil
+            end
+          }.to create_an_aws_network_acl('test_network_acl',
+            vpc_id: test_vpc.aws_object.id,
+            entry_set:
+              [
+                { :rule_number=>500, :protocol=>"-1", :rule_action=>"allow", :egress=>true, :cidr_block=>"0.0.0.0/0" },
+                { :rule_number=>32767, :protocol=>"-1", :rule_action=>"deny", :egress=>true, :cidr_block=>"0.0.0.0/0" },
+                { :rule_number=>100, :protocol=>"-1", :rule_action=>"deny", :egress=>false, :cidr_block=>"10.0.0.0/24" },
+                { :rule_number=>32767, :protocol=>"-1", :rule_action=>"deny", :egress=>false, :cidr_block=>"0.0.0.0/0" }
+              ]
+          ).and be_idempotent
+        end
```

Comment:
I need some help on this test. Everything works (it's definitely checking to make sure the rules match) but the test fails because `.updated?` is false, not true. Is there an easy way to tell the test that's what we want?